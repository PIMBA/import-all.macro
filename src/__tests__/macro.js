import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-macros'

const projectRoot = path.join(__dirname, '../../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    'no usage': `import importAll from '../macro'`,
    'import all files': `
      import importAll from '../macro'

      const a = importAll('./fixtures/*.js')
    `,
    'importAll.async uses dynamic import': `
      import importAll from '../macro'

      document.getElementById('load-stuff').addEventListener(() => {
        importAll.async('./fixtures/*.js').then(all => {
          console.log(all)
        })
      })
    `,
  },
})
