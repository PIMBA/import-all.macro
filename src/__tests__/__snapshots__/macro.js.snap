// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import all files 1`] = `

import importAll from '../macro'

const a = importAll('./fixtures/*.js')

      ↓ ↓ ↓ ↓ ↓ ↓

import * as _a from './fixtures/a.js';
import * as _b from './fixtures/b.js';
import * as _c from './fixtures/c.js';
import * as _d from './fixtures/d.js';


const a = {
  'a': _a,
  'b': _b,
  'c': _c,
  'd': _d
};

`;

exports[`importAll.async uses dynamic import 1`] = `

import importAll from '../macro'

document.getElementById('load-stuff').addEventListener(() => {
  importAll.async('./fixtures/*.js').then(all => {
    console.log(all)
  })
})

      ↓ ↓ ↓ ↓ ↓ ↓

document.getElementById('load-stuff').addEventListener(() => {
  Promise.all([import('./fixtures/a.js'), import('./fixtures/b.js'), import('./fixtures/c.js'), import('./fixtures/d.js')]).then(function importAllHandler(importVals) {
    return {
      'a': importVals[0],
      'b': importVals[1],
      'c': importVals[2],
      'd': importVals[3]
    };
  }).then(all => {
    console.log(all);
  });
});

`;

exports[`no usage 1`] = `

import importAll from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;
